src/scratchOrgSettingsGenerator.ts

```diff
@@ -148,8 +143,14 @@ export default class SettingsGenerator {
     const status = (await client.subscribe()) as string;
 
     if (status !== RequestStatus.Succeeded) {
+      const componentFailures = ensureObject<{
+        componentFailures: Record<string, unknown> | Array<Record<string, unknown>>;
+      }>(result.details).componentFailures;
+      const failures = (Array.isArray(componentFailures) ? componentFailures : [componentFailures])
+        .map((failure) => failure.problem)
+        .join('\n');
       const error = new SfdxError(
-        `A scratch org was created with username ${username}, but the settings failed to deploy`,
+        `A scratch org was created with username ${username}, but the settings failed to deploy due to: \n${failures}`,
```

Comment:
good for windows?  Or do we need `os.EOL` ?

Reply:
pretty sure `\n` is fine, it's used in other places in core?